pyproject.toml

```diff
@@ -31,6 +31,8 @@ hvac = { version = ">=0.11.0, <1.1.0", optional = true }
 nautobot = "^2.0.0"
 python = ">=3.8,<3.12"
 python-tss-sdk = {version = "~1.2.0", optional = true}
+azure-identity = "^1.15.0"
+azure-keyvault-secrets = "^4.8.0"
```

Comment:
These should be added as `optional = true` like the other provider-specific dependencies.